rfml.go

```diff
@@ -476,6 +464,120 @@ func uploadRFMLFilesInDirectory(rfmlDirectory string) error {
 	return nil
 }
 
+func downloadRFML(c cliContext) error {
+	testDirectory := c.String("test-folder")
+	absTestDirectory, err := prepareTestDirectory(testDirectory)
+	if err != nil {
+		return cli.NewExitError(err.Error(), 1)
+	}
+
+	var mappings rainforest.TestIDMappings
+	mappings, err = api.GetRFMLIDs()
+	if err != nil {
+		return cli.NewExitError(err.Error(), 1)
+	}
+
+	var testIDs []int
+	if len(c.Args()) > 0 {
+		var testID int
+		for _, arg := range c.Args() {
+			testID, err = strconv.Atoi(arg)
+			if err != nil {
+				return cli.NewExitError(err.Error(), 1)
+			}
+
+			testIDs = append(testIDs, testID)
+		}
+	} else {
+		for _, testIDMap := range mappings {
+			testID := testIDMap.ID
+			testIDs = append(testIDs, testID)
+		}
+	}
+
+	errorsChan := make(chan error)
+	testIDChan := make(chan int, len(testIDs))
+	testChan := make(chan *rainforest.RFTest, len(testIDs))
+
+	for _, testID := range testIDs {
+		testIDChan <- testID
+	}
+	close(testIDChan)
+
+	for i := 0; i < rfmlDownloadConcurrency; i++ {
+		go downloadRFTestWorker(testIDChan, errorsChan, testChan)
+	}
+
+	for i := 0; i < len(testIDs); i++ {
+		select {
+		case err = <-errorsChan:
+			return err
+		case test := <-testChan:
+			test.UnmapBrowsers()
+			err = test.UnmarshalElements(mappings)
+			if err != nil {
+				return err
+			}
+
+			paddedTestID := fmt.Sprintf("%010d", test.TestID)
+			sanitizedTitle := strings.TrimSpace(test.Title)
+			fileName := fmt.Sprintf("%v_%v.rfml", paddedTestID, sanitizedTitle)
+			rfmlFilePath := filepath.Join(absTestDirectory, fileName)
+
+			var file *os.File
+			file, err = os.Create(rfmlFilePath)
+			if err != nil {
+				return err
+			}
+
+			writer := rainforest.NewRFMLWriter(file)
+			err = writer.WriteRFMLTest(test)
+			file.Close()
+			if err != nil {
+				return err
+			}
+
+			log.Printf("Downloaded RFML test to %v", rfmlFilePath)
+		}
+	}
+
+	return nil
+}
+
+func downloadRFTestWorker(testIDChan chan int, errorsChan chan error, testChan chan *rainforest.RFTest) {
+	for testID := range testIDChan {
+		test, err := api.GetTest(testID)
+		if err != nil {
+			errorsChan <- err
+			return
+		}
+		testChan <- test
+	}
+}
+
+/*
+	Helper Functions
+*/
+
+func prepareTestDirectory(testDir string) (string, error) {
+	absTestDirectory, err := filepath.Abs(testDir)
+	if err != nil {
+		return "", cli.NewExitError(err.Error(), 1)
```

Comment:
nuke `NewExitError` from here, as we already do it from upstream: https://github.com/rainforestapp/rainforest-cli/pull/236/files#diff-0dae6029cce0003a807265a154d65843R471